plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/AbstractASTManager.java

```diff
@@ -995,6 +1015,41 @@ public TokensList getCompletionsFromTokenInLocalScope(IModule module, ICompletio
         return null;
     }
 
+    @Override
+    public boolean isNodeTypingUnionSubscript(IModule module, Object node) {
+        if (node instanceof Subscript) {
+            Subscript subscript = (Subscript) node;
+
+            String rep = NodeUtils.getFullRepresentationString(subscript);
```

Comment:
This seems a bit strange to me, why don't you just check the subscript value to know if it's what you expect?